server/vulnerabilities/msrc/parsed/full_product_name.go

```diff
@@ -0,0 +1,76 @@
+package msrc_parsed
+
+import "strings"
+
+// FullProdName abstracts a MS full product name.
+// A full product name includes the name of the product plus its arch
+// (if any) and its version (if any).
+type FullProdName string
+
+func NewFullProductName(fullName string) FullProdName {
+	return FullProdName(fullName)
+}
+
+// Arch returns the archicture from a Microsoft full product name, if none can
+// be found then "all" is returned.
+// eg:
+// "Windows 10 Version 1803 for 32-bit Systems" => "32"
+func (pn FullProdName) Arch() string {
+	val := string(pn)
+	switch {
+	case strings.Index(val, "32-bit") != -1:
+		return "32-bit"
+	case strings.Index(val, "x64") != -1:
+		return "64-bit"
+	case strings.Index(val, "ARM64") != -1:
+		return "arm64"
```

Comment:
is there any reason to use `"32-bit"` and `"64-bit"` as the return values? `arm64` is also 64-bit. From the context I believe this is `x86` and `x86-64` right? maybe use the same values as `GOARCH` just to keep the same conventions?

Reply:
Those values are meant to match whatever you get from running `SELECT arch FROM os_version` in OSQuery:

![image](https://user-images.githubusercontent.com/8041318/187276077-10c07060-78dc-43b9-8ddc-01927ccee355.png)

I'll add a comment to make it more clear.

Reply:
ahh, makes sense! thanks